src/main/kotlin/com/sourcegraph/cody/edit/HistoryManager.kt

```diff
@@ -0,0 +1,37 @@
+package com.sourcegraph.cody.edit
+
+/** Manages user prompt history in memory for a session (project lifetime). */
+class HistoryManager<T>(private val capacity: Int) {
```

Comment:
At some point we could maybe unify this and `CodyChatMessageHistory`

Reply:
Maybe! We'll have to see how similar they are in practice. But I do think we should add persistence to the history for Inline Edits, so that may be a good solution.